packages/wonder-blocks-date-picker/src/util/temporal-locale-utils.ts

```diff
@@ -0,0 +1,395 @@
+import {Temporal} from "temporal-polyfill";
+import {CustomModifiers} from "./types";
+
+/**
+ * Utility functions for working with Temporal dates in react-day-picker.
+ * These replace the MomentLocaleUtils that were previously used.
+ *
+ * Question: Should we move this to a separate package?
+ */
+
+/**
+ * Format a Temporal.PlainDate using a format string.
+ * Supports a subset of moment.js format tokens for compatibility.
+ */
+export function formatDate(
+    date: Temporal.PlainDate,
+    format: string | Array<string> | null | undefined,
+    locale: string = "en",
+): string {
```

Comment:
What happens if a date can't be formatted? Is an error thrown or is the default ISO used?